google/auth/_default.py

```diff
@@ -0,0 +1,295 @@
+# Copyright 2015 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Application default credentials.
+
+Implementes application default credentials and project ID detection."""
+
+import io
+import json
+import os
+
+from six.moves import configparser
+
+from google.auth import compute_engine
+from google.auth import exceptions
+from google.auth.compute_engine import _metadata
+import google.auth.transport._http_client
+from google.oauth2 import service_account
+import google.oauth2.credentials
+
+# Environment variable for explicit application default credentials and project
+# ID.
+_CREDENTIALS_ENV = 'GOOGLE_APPLICATION_CREDENTIALS'
+_PROJECT_ENV = 'GCLOUD_PROJECT'
+
+# Valid types accepted for file-based credentials.
+_AUTHORIZED_USER_TYPE = 'authorized_user'
+_SERVICE_ACCOUNT_TYPE = 'service_account'
+_VALID_TYPES = (_AUTHORIZED_USER_TYPE, _SERVICE_ACCOUNT_TYPE)
+
+# The Google OAuth 2.0 token endpoint. Used for authorized user credentials.
+_GOOGLE_OAUTH2_TOKEN_ENDPOINT = 'https://accounts.google.com/o/oauth2/token'
+
+# The ~/.config subdirectory containing gcloud credentials.
+_CLOUDSDK_CONFIG_DIRECTORY = 'gcloud'
+# The environment variable name which can replace ~/.config if set.
+_CLOUDSDK_CONFIG_ENV = 'CLOUDSDK_CONFIG'
+# The name of the file in the Cloud SDK config that contains default
+# credentials.
+_CLOUDSDK_CREDENTIALS_FILENAME = 'application_default_credentials.json'
+# The name of the file in the Cloud SDK config that contains the
+# active configuration.
+_CLOUDSDK_ACTIVE_CONFIG_FILENAME = os.path.join(
+    'configurations', 'config_default')
+# The config section and key for the project ID in the cloud SDK config.
+_CLOUDSDK_PROJECT_CONFIG_SECTION = 'core'
+_CLOUDSDK_PROJECT_CONFIG_KEY = 'project'
+
+# Help message when no credentials can be found.
+_HELP_MESSAGE = (
+    'Could not automatically determine credentials. Please set {env} or '
+    'explicitly create credential and re-run the application. For more '
+    'information, please see https://developers.google.com/accounts/docs'
+    '/application-default-credentials.'.format(env=_CREDENTIALS_ENV))
```

Comment:
Would you prefer me to break at 80 lines or just the terminal's let screen wrapping handling it?